core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java

```diff
@@ -282,20 +283,77 @@ private static boolean isCreate(UpdateTableRequest request) {
     return isCreate;
   }
 
-  private static TableMetadata create(TableOperations ops, TableMetadata start, UpdateTableRequest request) {
+  private static TableMetadata commitUpdate(Catalog catalog, TableIdentifier ident, UpdateTableRequest request) {
+    Table table = catalog.loadTable(ident);
+    if (table instanceof BaseTable) {
+      TableOperations ops = ((BaseTable) table).operations();
+      return commit(ops, request.requirements(), request.updates());
+    } else {
+      throw new IllegalStateException("Cannot wrap catalog that does not produce BaseTable");
+    }
+  }
+
+  private static TableMetadata commitCreate(Catalog catalog, TableIdentifier ident, UpdateTableRequest request) {
+    // this is a hacky way to get TableOperations for an uncommitted table
+    Transaction transaction = catalog.buildTable(ident, EMPTY_SCHEMA).createOrReplaceTransaction();
```

Comment:
I think this should just build the transaction directly, instead of starting with a `newTableMetadata`:

```java
CreateCommitInfo info = extractCreateCommitInfo(request);
TableBuilder builder = catalog.buildTable(info.schema)
if (info.unboundPartitionSpec != null) {
  builder.withPartitionSpec(info.unboundPartitionSpec.bind(info.schema);
}
...
Transaction transaction = builder.createOrReplaceTransaction();
TableOperations ops = ((TransactionTable) transaction.table()).operations();
TableMetadata start = ops.current();
TableMetadata.Builder metaBuilder = TableMetadata.buildFrom(start)
info.oherUpdates.forEach(update -> update.applyTo(metaBuilder));
ops.commit(start, builder.build());
```

Then we should be able to remove the `underlyingOps` and `startMetadata` methods that were added to make this work!

Reply:
I attempted this approach but it wasn't working for me. The ops looks like it was a temporary ops that didn't perform a commit to the delegate catalog.

Reply:
Ah, yeah. You'd probably need to use `commitTransaction()` at the end.

Reply:
I made this change, though still access `underlyingOps()` to check the requirements. Alternatively I could call `builder.createTransaction()` which will throw the same exception, but the message is different (and causes a test assertion to fail).

Reply:
I still need to fix something, also I'm adding a test...

Reply:
I made this change. Filtering out the setter types with ID of -1 wasn't working, I changed that to filter those out only if an associated add type is pending. I also added a test that adds 2 files, each associated with a different schema, spec, and sort order.